app/controllers/staff/speakers_controller.rb

```diff
@@ -1,60 +1,59 @@
 class Staff::SpeakersController < Staff::ApplicationController
   decorates_assigned :speaker
-  before_filter :set_proposal, only: [:new, :create, :destroy]
+  before_action :set_program_session, only: [:new, :create]
 
   def index
-    render locals: {
-             proposals: @event.proposals.includes(speakers: :user).decorate
-           }
+    @program_speakers = current_event.speakers.in_program
   end
 
   def new
     @speaker = Speaker.new
-    @user = @speaker.build_user
-    @proposal = Proposal.find_by(uuid: params[:proposal_uuid])
   end
 
   def create
     s_params = speaker_params
-    user = User.find_by(email: s_params.delete(:speaker_email))
-    if user
-      if user.speakers.create(s_params.merge(proposal: @proposal))
-        flash[:success] = "Speaker was added to this proposal"
-      else
-        flash[:danger] = "There was a problem saving this speaker"
-      end
+    @speaker = @program_session.speakers.create(s_params.merge(event: current_event))
+    authorize @speaker
+    if @speaker.save
+      flash[:success] = "#{@speaker.name} has been added to #{@program_session.title}"
+      redirect_to event_staff_speakers_path(current_event)
     else
-      flash[:danger] = "Could not find a user with this email address"
+      flash[:danger] = "There was a problem saving this speaker."
+      render :new
     end
-    redirect_to event_staff_proposal_path(event, @proposal)
   end
 
-  #if user input (params), exist
-
   def show
-    @speaker = Speaker.find(params[:id])
+    @speaker = current_event.speakers.find(params[:id])
   end
 
   def edit
-    @speaker = Speaker.find(params[:id])
+    @speaker = current_event.speakers.find(params[:id])
   end
 
   def update
-    @speaker = Speaker.find(params[:id])
+    @speaker = current_event.speakers.find(params[:id])
+    authorize @speaker
     if @speaker.update(speaker_params)
-      redirect_to event_staff_speaker_url(@event, @speaker)
+      flash[:success] = "#{@speaker.name} was successfully updated"
+      redirect_to event_staff_speakers_path(current_event)
     else
+      flash[:danger] = "There was a problem updating this speaker."
       render :edit
     end
   end
 
   def destroy
-    @speaker = Speaker.find_by!(id: params[:id])
-    proposal = speaker.proposal
-    @speaker.destroy
-
-    flash[:info] = "You've deleted the speaker for this proposal"
-    redirect_to event_staff_proposal_path(uuid: proposal)
+    @speaker = current_event.speakers.find(params[:id])
+    authorize @speaker
+    if current_event.speakers.count > 1
+      @speaker.destroy
```

Comment:
Not essential, but destroy may fail here and the code will still display a success message. If you decide to fix though, this method is getting kinda busy, so I'd recommend moving the speaker count check into a before_action first to avoid clutter.